Streams/FileSliceReader.cpp

```diff
@@ -42,6 +42,13 @@ void FileSliceReader::ReadImplementation(void* buffer, std::size_t size)
 	fileStreamReader.Read(buffer, size);
 }
 
+std::size_t FileSliceReader::ReadPartial(void* buffer, std::size_t size) {
+	auto bytesLeft = sliceLength - (fileStreamReader.Position() - startingOffset);
```

Comment:
Any reason not to make this line read:

```auto bytesLeft = sliceLength - Position();```

I think this change would read a bit easier.